kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/StartsWithExpressionEvaluator.java

```diff
@@ -46,7 +49,20 @@ static Predicate validateAndTransform(
         childrenExpressions.get(1),
         startsWith,
         "'STARTS_WITH' expects literal as the second input");
-    return new Predicate(startsWith.getName(), childrenExpressions);
+
+    if (startsWith.getCollationIdentifier().isPresent()) {
+      CollationIdentifier collationIdentifier = startsWith.getCollationIdentifier().get();
+      if (!collationIdentifier.isSparkUTF8BinaryCollation()) {
+        String msg =
+            format(
+                "Unsupported collation: \"%s\". Default Engine supports just"
+                    + " \"%s\" collation.",
+                collationIdentifier, CollationIdentifier.SPARK_UTF8_BINARY);
+        throw unsupportedExpressionException(startsWith, msg);
+      }
+    }
```

Comment:
nit: please extract the check for supported collation into an utility and reuse it.